R/render.R

```diff
@@ -343,124 +356,75 @@ add_links <- function(path, main_language = main_language,
     if (no_translated_version) return()
   }
 
-  if (type == "book") {
+  languages_links <- xml2::xml_find_first(html, "//ul[@id='languages-links']")
+  languages_links_div_exists <- (length(languages_links) > 0)
 
-    sidebar_menu <- xml2::xml_find_first(html, "//div[contains(@class,'sidebar-menu-container')]")
+  if (!languages_links_div_exists) {
+    if (placement == "navbar") {
+      navbar <- xml2::xml_find_first(html, "//div[@id='navbarCollapse']")
```

Comment:
could someone set their navbar so that there's no collapse though? https://quarto.org/docs/websites/website-navigation.html#top-navigation

Reply:
Plus the collapse can be either on the left or right? So should the navbar node used here be the last child of the navbar div or something like that?

Reply:
I tried to make it much more robust by looking for a better place in the navbar. I think the new code should cover all the different options.